.github/workflows/release.yaml

```diff
@@ -71,7 +74,7 @@ jobs:
         # Get the latest amd64 revision in CharmHub and increments it by one.
         # This assumes that the amd64 and arm64 revisions are in lockstep.
         run: |
-          echo "::set-output name=rev::$(($(charmcraft resource-revisions prometheus-k8s promql-transform-amd64 | tail -n+2 | head -1 | awk '{ print $1 }') + 1))"
+          echo "::set-output name=rev::$(($(charmcraft resource-revisions ${{ matrix.charm }} promql-transform-amd64 | tail -n+2 | head -1 | awk '{ print $1 }') + 1))"
```

Comment:
Iiuc, `rev` is used here for tagging releases in _this_ repo, however:
1. A given charm can have different resource revisions for amd64 and arm64, but sticking with just one of them is fine, except -
2. prometheus-k8s and grafana-k8s and future additions would all have different resource revision numbers for the same resource. Would multiple `rev` tags be created?
3. A race condition between this and manual cli is technically possible. Can we instead parse the output of the `upload-resource` command?

Reply:
(1) is true, we should use a matrix for the arch too and that would solve itself.
(2) creating multiple rev tags is fine, although not optimal, as long as the tag has the arch in the tag name.
(3) Problem not worth solving IMO.